tests/testthat/test-duplicate_argument_linter.R

```diff
@@ -95,3 +97,65 @@ test_that("doesn't lint duplicated arguments in allowed functions", {
     linter
   )
 })
+
+test_that("interceding comments don't trip up logic", {
+  linter <- duplicate_argument_linter()
+  lint_msg <- rex::rex("Avoid duplicate arguments")
+
+  # actually this case "just works" even before #2402 since
+  #   get_r_string() returns NA for both argument names
+  expect_lint(
+    trim_some("
+      fun(
+        arg # xxx
```

Comment:
Sure! The current bug definitely only needed the earlier tests, but it's good to have further robustness tests in case we refactor later. Done now.

Also: https://github.com/r-lib/lintr/issues/2191#issuecomment-1846513844